sdk/core/azure-core/src/main/java/com/azure/core/implementation/AccessTokenCache.java

```diff
@@ -161,6 +170,83 @@ private Supplier<Mono<? extends AccessToken>> retrieveToken(TokenRequestContext
         };
     }
 
+    private Supplier<AccessToken> retrieveTokenSync(TokenRequestContext tokenRequestContext,
+                                                                boolean checkToForceFetchToken) {
+        return () -> {
+            if (tokenRequestContext == null) {
+                throw LOGGER.logExceptionAsError(
+                    new IllegalArgumentException("The token request context input cannot be null."));
+            }
+            AccessTokenCacheInfo cache = this.cacheInfo.get();
+            AccessToken cachedToken = cache.getCachedAccessToken();
+
+            OffsetDateTime now = OffsetDateTime.now();
+            Supplier<AccessToken> tokenRefresh;
+            AccessToken fallback;
+
+            // Check if the incoming token request context is different from the cached one. A different
+            // token request context, requires to fetch a new token as the cached one won't work for the
+            // passed in token request context.
+            boolean forceRefresh = (checkToForceFetchToken && checkIfForceRefreshRequired(tokenRequestContext))
+                || this.tokenRequestContext == null;
+
+            Supplier<AccessToken> tokenSupplier = () ->
+                tokenCredential.getTokenSync(this.tokenRequestContext);
```

Comment:
We never have another implementation of this, so can it be static to avoid re-creating a lambda?

Reply:
it uses a constructor parameter, which is tied to the instance, at best we can make this a class instance var to avoid re-creating it.